docs/constructive.rst

```diff
@@ -1,9 +1,11 @@
 Constructive operations
 =======================
 
-.. automodule:: shapely.constructive
-   :members:
-   :exclude-members: BufferCapStyles, BufferJoinStyles, minimum_rotated_rectangle
-   :special-members:
-   :inherited-members:
-   :show-inheritance:
+.. currentmodule:: shapely
+
+.. autosummary::
+   :toctree: reference/
+
+{% for function in get_module_functions("constructive") %}
+   {{ function }}
+{% endfor %}
```

Comment:
Alternatively to this templating, we could also just hardcode this list here, which would look like:

```
.. autosummary::
   :toctree: reference/

   boundary
   buffer
   offset_curve
   centroid
   clip_by_rect
   convex_hull
   delaunay_triangles
   segmentize
   envelope
   extract_unique_points
   build_area
   make_valid
   normalize
   point_on_surface
   polygonize
   polygonize_full
   reverse
   simplify
   snap
   voronoi_polygons
   oriented_envelope
   minimum_rotated_rectangle
   minimum_bounding_circle
```

That's easy to do once. But afterwards, we will have to remember to add an entry here when adding a new function (also not that much of work, so maybe this is not worth the complexity of the templating)
```